ONSdigital/ssdc-rm-support-tool
# Add JS format check

# Motivation and Context
We want consistent and easy formatting of our JS files

# What has changed
- Add prettier as npm dev dependency
- Add make targets for formatting and format checking
- Add Makefile in UI sub-project
- Add a parallel Travis job with the node language for the UI checks

# How to test?
`make format` should format all files when run from the project root, and just the JS files when run inside the UI sub-project

# Using Prettier in VS Code and Webstorm
This PR allows formatting from the command line with `make format` but prettier also has support in both our IDEs of choice
- Webstorm https://www.jetbrains.com/help/webstorm/prettier.html#ws_prettier_reformat_code (note you should not need to do the "installation" steps as we have prettier installed as a project dependency if you open the `./ui` sub project directly)
- VS Code https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

# Links
https://trello.com/c/YAaiYJTS/2551-add-format-checker-to-support-tool-javascript
